### PR TITLE
Introduce `pre-commit`

### PR DESCRIPTION
**Pull request by yu-iskw**
## Overview
In my opinion, it would be good to introduce pre-commit to develop dbt-osmosis so that we consistently format files and coding standards a head of pushing the default branch. 

We can run the hooks with the subsequent commands. 
```
pip install -U pre-commit

pre-commit install

pre-commit run --all-files
```

### CODE REVIEW

**Comment by yu-iskw** [comment]
@z3z1ma Can you review the pull request?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#      - id: isort
+#  - repo: https://github.com/pycqa/flake8
+#    rev: 4.0.1
+#    hooks:
+#      - id: flake8
+# TODO refactor the files in the 'docker' directory later.
+#  - repo: https://github.com/hadolint/hadolint
+#    rev: v2.12.0
+#    hooks:
+#      - id: hadolint-docker
```
[the remark this answers is not in the document]
Comment (z3z1ma) [reply .pre-commit-config.yaml:36]: Agreed, this can be a follow up.

**Comment by z3z1ma** [comment]
Same as other.
Once we rebase and resolve conflicts, I will merge and cut a release. Good call on pre-commit.

**Comment by yu-iskw** [comment]
I will resolve the conflicts as soon as possible.

**Comment by yu-iskw** [comment]
@z3z1ma Can you review it again? I have resolved the conflicts. To do so, I had to get `poetry.lock` updated with `poetry lock --no-update`. But, I don't think it breaks anything in the existing functions.

**Pull request merged**